src/components/form/PasswordInputField.tsx

```diff
@@ -0,0 +1,58 @@
+import IconButton from "@mui/material/IconButton";
+import VisibilityIcon from "@mui/icons-material/Visibility";
+import VisibilityOffIcon from "@mui/icons-material/VisibilityOff";
+import { useTheme } from "@mui/material";
+import { FunctionComponent, useState } from "react";
+import { TextInputField } from "components";
+
+interface PasswordInputFieldProps {
+  readonly ariaLabel?: string;
+  readonly endAdornmentHandler?: () => void;
+  readonly externalMaskPassword?: boolean;
+  readonly name: string;
+  readonly placeholder?: string;
+  readonly showEndAdornment?: boolean;
+}
```

Comment:
I'd probably just extend the `TextInputProps`, that way you don't have to define props such as the `ariaLabel` (I think it's also fine to not worry about setting a default for it). That way it will have all of the props of a normal text input defined, in addtion to the password field props. E.g.

```
import { TextInputProps } from "elements";

...

interface PasswordInputFieldProps extends TextInputProps {
  readonly handlePressEndAdornment?: VoidFunction; // just preference, either name works
  readonly externalMaskPassword?: boolean;
  readonly showEndAdornment?: boolean;
}
```

Reply:
I thought about doing it this way but i saw an error in the console that i thought happened when spreading props:
![image](https://user-images.githubusercontent.com/11530120/166616260-13ddc340-5e02-4d1f-a5aa-09d25bd14ccf.png)

I applied the changes though

Reply:
Interesting. Yeah, that may be the case. I still think extending the props is good though, that way the user can use all of the other text input props, such as `autoComplete`, etc...